src/react-navigation/adapter.ts

```diff
@@ -0,0 +1,23 @@
+import {
+  // @ts-ignore: this hook is not available in React Navigation v7
+  useLinkBuilder,
+  // @ts-ignore: this hook is not available in React Navigation v6
+  useLinkTools,
+} from '@react-navigation/native';
+
+function useV6Link() {
+  return useLinkBuilder();
+}
+
+function useV7Link() {
+  const tools = useLinkTools();
+  return tools.buildHref;
+}
+
+type NavigationLink = () => (
+  name: string,
+  params?: object
+) => string | undefined;
+
+export const useNavigationLink: NavigationLink =
+  typeof useLinkTools !== 'undefined' ? useV7Link : useV6Link;
```

Comment:
I'd remove `useV6Link`
```suggestion
export const useNavigationLink: NavigationLink =
  typeof useLinkTools !== 'undefined' ? useV7Link : useLinkBuilder;
```
Also can you add a comment:
> In React Navigation 7 `useLinkBuilder` was superseded by `useLinkTools`
> https://reactnavigation.org/docs/7.x/upgrading-from-6.x/#the-uselinkto-and-uselinkbuilder-hooks-are-merged-into-uselinktools

Reply:
Thanks, updated